### ISSUE TITLE
Sites that are blocked before we can check them for DNT are never unblocked.

### ISSUE THREAD

**Issue by cowlicks** [EFForg/privacybadger#1640]
This happens because we block requests in `onBeforeRequest`, but check for DNT in `onBeforeSendHeaders`.

To test this visit:
https://necessaryandproportionate.org/sign

Then run this in your console:
```javascript
badger.storage.getBadgerStorageObject('action_map').setItem('eff.org', {dnt: false, heuristicAction: "block", nextUpdateTime: 0, userAction: ""})
```

Refresh https://necessaryandproportionate.org/sign you'll see the eff domain is blocked. And even though `nextUpdateTime` is 0, it won't get rechecked.

**Comment by ghostwords** [comment]
Duplicate of #1398?

**Comment by cowlicks** [comment]
@ghostwords yep. closing. I'm working on a fix

**Comment by ghostwords** [comment]
It would be good to know how to reproduce it without setting Badger state in the debugger.
